### PR TITLE
make it possible to show the checks of more than one team

### PR DESCRIPTION
**Pull request by ulich**
Before:
image

After:
image

### CODE REVIEW

File: dash-core/src/main/resources/static/dash.js [hunk 1]
```diff
@@ -33,6 +39,14 @@ angular.module('dash', ['ngResource', 'ngSanitize'])
 
         teamsResource.get().$promise.then(function (teams) {
             $scope.teams = teams['teams'];
+
+            // if config.teams is an empty object, initialize it to show all teams
+            // (this is only the case when the team checkboxes have never been touched)
+            if (Object.getOwnPropertyNames($scope.config.teams).length == 0) {
```
Comment (gaffa) [inline dash-core/src/main/resources/static/dash.js]: is this really necessary or can this be handled a more elegant way? this way of initialization looks a little "hmm". :)

File: dash-core/src/main/resources/static/dash.js [hunk 2]
```diff
@@ -223,21 +237,14 @@ angular.module('dash', ['ngResource', 'ngSanitize'])
             return 4 - scoreForState(state);
         };
 
-        $scope.teamFilter = function (check) {
+        $scope.teamFilter = showCheckForChosenTeams;
 
-            if ($scope.config.team == "All Teams") {
+        function showCheckForChosenTeams(check) {
```
Comment (gaffa) [inline dash-core/src/main/resources/static/dash.js]: i think the naming is not optimal.

File: dash-core/src/main/resources/static/dash.js (hunk 1 above)
Comment (ulich) [inline dash-core/src/main/resources/static/dash.js]: There is no good way of checking for an empty object in javascript... I could set the default value of $scope.config.teams to null or so, but then I would have to add a check for null to several places

**Comment by ulich** [comment]
I made the config.teams inizalization code more clear

**Pull request merged**